pkg/frontend/kubeactions/kubeactions.go

```diff
@@ -160,6 +163,63 @@ func (ka *kubeactions) List(ctx context.Context, oc *api.OpenShiftCluster, kind,
 	return ul.MarshalJSON()
 }
 
+func (ka *kubeactions) applyOne(ctx context.Context, dyn dynamic.Interface, grs []*restmapper.APIGroupResources, un *unstructured.Unstructured) error {
+	if strings.EqualFold(un.GetKind(), "secret") { // TODO do we need this?
+		return fmt.Errorf("access to secrets is forbidden")
+	}
+	gvrs := ka.findGVR(grs, un.GetKind())
+	if len(gvrs) == 0 {
+		return fmt.Errorf("Kind %s not found", un.GetKind())
+	}
+	if len(gvrs) > 1 {
+		return fmt.Errorf("Kind %s did not uniquely match one GRV", un.GetKind())
+	}
+	gvr := gvrs[0]
+
+	ka.log.Debugf("appling kube object %s %s %s ", un.GetKind(), un.GetNamespace(), un.GetName())
+
+	_, err := dyn.Resource(*gvr).Namespace(un.GetNamespace()).Create(un, metav1.CreateOptions{})
+	if err == nil || !errors.IsAlreadyExists(err) {
+		return err
+	}
+
+	return retry.RetryOnConflict(retry.DefaultRetry, func() error {
+		_obj, err := dyn.Resource(*gvr).Namespace(un.GetNamespace()).Get(un.GetName(), metav1.GetOptions{})
+		if err != nil {
+			return err
+		}
+
+		un.SetResourceVersion(_obj.GetResourceVersion())
+		_, err = dyn.Resource(*gvr).Namespace(un.GetNamespace()).Update(un, metav1.UpdateOptions{})
+		return err
+	})
```

Comment:
a rare occasion where RetryOnConflict should not be used.  Try the update once, and if it fails, return the error and let the user decide what to do.  Otherwise the following could happen:

1. SRE gets object
2. Cluster user updates object
3. SRE updates object, losing changes from 2